NVSHAS-9494 make manager SLSA capable

holyspectral opened the pull request:
Create release pipeline and standalone Dockerfile based on https://github.com/neuvector/manager/pull/765

NickChungSUSE reviewed (approved):
LGTM! It works on my side.

pjbgf commented on package/Dockerfile:
```diff
@@ -0,0 +1,107 @@
+# Manager unitest
+RUN if [ "$ARCH" = "amd64" ]; then \
+  zypper addrepo https://download.opensuse.org/repositories/M17N:fonts/15.6/M17N:fonts.repo && \
+  zypper --non-interactive --gpg-auto-import-keys refresh && \
```
Instead of the `--gpg-auto-import-keys`. The specific repo key can be imported ahead of time, IIRC this should work `rpm --import https://download.opensuse.org/repositories/M17N/15.6/repodata/repomd.xml.key`.

pjbgf commented on package/Dockerfile:
```diff
@@ -0,0 +1,107 @@
+COPY --from=builder /src/package/requirements.txt /chroot/requirements.txt
+COPY --from=micro / /chroot/
+
+RUN zypper --installroot /chroot -n --gpg-auto-import-keys in --no-recommends \
```
Can we avoid `--gpg-auto-import-keys`?

holyspectral commented on package/Dockerfile:
```diff
@@ -0,0 +1,107 @@
+    zypper install -y ca-certificates wget curl zip git awk nodejs20 npm20
+
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+    curl -fL https://github.com/coursier/launchers/raw/master/cs-x86_64-pc-${TARGETOS}.gz | gzip -d > cs; \
+  elif [ "$TARGETARCH" = "arm64" ]; then \
+    curl -fL https://github.com/VirtusLab/coursier-m1/releases/latest/download/cs-aarch64-pc-${TARGETOS}.gz | gzip -d > cs; \
```
I will pin it to c2eb9a75082b4f70b12104b6da74cfff3191dbcd and [v2.1.17](https://github.com/VirtusLab/coursier-m1/releases/tag/v2.1.17). 
@kyledong-suse @NickChungSUSE could you please help to update neuvector/manifests accordingly?

holyspectral replied:
Moved to 2.1.18.

holyspectral commented:
Rebased and implemented review comments in https://github.com/neuvector/manager/pull/767/commits/2bdbaef17230603f9bb93b52b9530bc724a26867

@pjbgf @NickChungSUSE please take a look.

pjbgf commented on package/Dockerfile:
```diff
@@ -0,0 +1,118 @@
+    zypper install -y ca-certificates wget curl zip git awk nodejs20 npm20
+
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+    curl -fL https://github.com/coursier/launchers/raw/c2eb9a75082b4f70b12104b6da74cfff3191dbcd/cs-x86_64-pc-${TARGETOS}.gz | gzip -d > cs; \
+  elif [ "$TARGETARCH" = "arm64" ]; then \
+    curl -fL https://github.com/VirtusLab/coursier-m1/releases/download/v2.1.17/cs-aarch64-pc-${TARGETOS}.gz | gzip -d > cs; \
```
To make automation easier (for auto bumps), the version/commit can be extracted to its own variable. However, this is personal preference, feel free to keep as is.

pjbgf commented on package/build_manager.sh:
```diff
@@ -0,0 +1,59 @@
+        -d)
+        mkdir -p /root/.ivy2
+        ln -s /prebuild/manager/cache /root/.ivy2/cache
+        # ln -s /prebuild/manager/node_modules node_modules
```
Is this no longer needed?

NickChungSUSE replied:
Actually, this line was commented out a long time ago, and we do have the npm install script below for the installation of node_modules, so I think this line can be removed.

pjbgf commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,60 @@
+      uses: docker/login-action@v3
+      with:
+        registry: docker.io
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
```
Following this login, a new tag is created for `neuvector/manager` from `neuvector/neuvector-manager`, which is the same org as above:

```suggestion
        username: ${{ env.DOCKER_USERNAME }}
        password: ${{ env.DOCKER_PASSWORD }}
```

holyspectral replied:
per our discussion, it's removed and we will call push-image twice.

holyspectral commented on package/build_manager.sh:
```diff
@@ -0,0 +1,59 @@
+cp -r cli/prog ${STAGE_DIR}/usr/local/bin/
+cp scripts/* ${STAGE_DIR}/usr/local/bin/
+cp java.security ${STAGE_DIR}/usr/lib/jvm/java-17-openjdk/lib/security/java.security
+cp java.security ${STAGE_DIR}/usr/lib64/jvm/java-17-openjdk-17/conf/security/java.security
```
I see.  I think it's because the files in /src/stage get overwritten by its next line.  

```
COPY --from=builder /src/stage /
COPY --from=builder /usr/lib64/ /usr/lib64/
```
Any concern if I swap them?

NickChungSUSE replied:
I think we can try swapping them to see if it works correctly. We just need to make sure the packages that Java requires under the /usr/lib64/ are included.

pjbgf commented on Makefile:
```diff
@@ -31,4 +31,69 @@ jar:
 	@echo "Making $@ ..."
-	docker run --rm -ia STDOUT --name build -v prebuild_manager:/prebuild/manager -v $(CURDIR):/manager -w /manager --entrypoint ./make_jar.sh neuvector/build_manager:${BUILD_IMAGE_TAG}
\ No newline at end of file
+	docker run --rm -ia STDOUT --name build -v prebuild_manager:/prebuild/manager -v $(CURDIR):/manager -w /manager --entrypoint ./make_jar.sh neuvector/build_manager:${BUILD_IMAGE_TAG}
```
Is this need for testing and local development only?

pjbgf reviewed (approved):
There is an outstanding question around the `jar` target, apart from that LGTM.
